File: 2784_Check_if_Array_is_Good/solution.py
class Solution:
    def isGood(self, nums: List[int]) -> bool:
        n = len(nums)
        c = [0] * n
        for a in nums:
            if a >= n:
                return False
            if a < n - 1 and c[a] > 0:
                return False
            if a == n - 1 and c[a] > 1:
                return False
            c[a] += 1
        return True

File: 2784_Check_if_Array_is_Good/solution.c
bool isGood(int* nums, int n) {
    int* c = (int*)calloc(n, sizeof(int));
    for (int i = 0; i < n; i++) {
        int a = nums[i];
        if (a < 1 || a >= n) { free(c); return false; }
        if (a < n - 1 && c[a] > 0) { free(c); return false; }
        if (a == n - 1 && c[a] > 1) { free(c); return false; }
        c[a]++;
    }
    free(c);
    return true;
}
